packages/instant/src/containers/selected_asset_retry_button.tsx

```diff
@@ -0,0 +1,26 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+
+import { Action, actions } from '../redux/actions';
+
+import { RetryButton } from '../components/retry_button';
+
+export interface SelectedAssetRetryButtonProps {}
```

Comment:
I would remove this file and have all this logic be handled by `BuyOrderStateButton` and the connector.

Reply:
I originally had it designed like that, but it felt to me like the connector was getting a bit overloaded, it had to connect a lot of things:

**To determine which button to render:**
 `buyOrderState`

**For buy button:**
`assetBuyer `, `buyQuote `, `onBuyClick`, `onBuySuccess`, `onBuyFailure`

**For retry button:**
`onRetryClick`

So even if we are just rendering the retry button, we are still connecting all of these buy button functions.  Felt like too much going on, should be separated out. But, I do see that the current implementation has a bit of boilerplate and and adds a little indirection.  Thoughts?

Reply:
Yes, Option 2 is what we have right now.  I"m down to  move the component definition, but would like to keep the architecture of Option 2 as-is for now